frontend/src/main/web/src/components/TextInput.jsx

```diff
@@ -0,0 +1,185 @@
+import React, { Component, PropTypes } from 'react'
+import ReactDOM from 'react-dom'
+import TextareaAutosize from 'react-textarea-autosize'
+import { flattenThemeClasses } from '../utils/styleUtils'
+import { isEqual } from 'lodash'
+
+const classes = {
+  base: {
+    ap: 'Ap(n)',
+    bgc: 'Bgc(t)',
+    bdc: 'Bdc(neutral)',
+    bdw: 'Bdw(2px)',
+    bds: 'Bds(s)',
+    bdrs: 'Bdrs(rq)',
+    bxz: 'Bxz(bb)',
+    c: 'C(i)',
+    ff: 'Ff(inh)',
+    fw: 'Fw(inh)',
+    fz: 'Fz(inh)',
+    fs: 'Fs(inh)',
+    o: 'O(n)',
+    p: 'Px(rq) Py(re)',
+    ph: 'Ph(neutral)',
+    focus: {
+      bdc: 'Bdc(pri):f'
+    }
+  }
+}
+
+class TextInput extends Component {
```

Comment:
Needs doc. Seems like the main thing is that it will be a single- or multiple-line input depending on how it is configured.